src/docs/stan-reference/appendices.tex

```diff
@@ -663,35 +664,55 @@ \subsection{Expressions}
 \small
 \begin{Verbatim}[fontsize=\small]
 expressions ::= expression % ','
-expression ::= real_literal
-             | variable
-             | '{' expressions '}'
-             | expression `?` expression `:` expression
+
+expression ::= expression `?` expression `:` expression
              | expression infixOp expression
              | prefixOp expression
              | expression postfixOp
-             | expression '[' indexes ']'
-             | function_literal '(' ?expressions ')'
-             | function_literal '(' expression ?('|' expression % ',') ')'
-             | integrate_ode '(' function_literal (',' expression){6} ')'
-             | integrate_ode_rk45
-               '(' function_literal (',' expression){6|9} ')'
-             | integrate_ode_bdf
-               '(' function_literal (',' expression){6|9} ')'
-             | algebra_solver
-             '('function_literal (',' expression){4|7} ')'
-             | '(' expression ')'
+             | common_expression
+
+range_expression ::= range_expression mathInfixOp range_expression
+                   | prefixOp range_expression
+                   | range_expression postfixOp
+                   | range_expression '[' indexes ']'
+                   | common_expression
+
+common_expression ::= real_literal
+                    | variable
+                    | '{' expressions '}'
+                    | function_literal '(' ?expressions ')'
+                    | function_literal '(' expression ?('|' expression % ',') ')'
+                    | integrate_ode '(' function_literal (',' expression){6} ')'
+                    | integrate_ode_rk45
+                      '(' function_literal (',' expression){6|9} ')'
+                    | integrate_ode_bdf
+                      '(' function_literal (',' expression){6|9} ')'
+                    | algebra_solver
+                    '('function_literal (',' expression){4|7} ')'
+                    | '(' expression ')'
+
+prefixOp ::= '!' | '-' | '+' | '^'
+
+postfixOp ::= '\\''
+
+infixOp ::= mathInfixOp | logicalInfixOp
+
+mathInfixOp ::= ('+' | '-' | '*' | '/' | '%' | '\\' | '.*' | './')
```

Comment:
Thanks for writing these out.  There's also a backslash for left division.  There should be a complete list in the manual under the precedence table.

Reply:
Sorry, that was just sloppiness.

Reply:
Rather, it was trying to figure out exactly how many backslashes needed to escape things.